prefixset_iter_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build go1.23
+
+package netipds
+
+import (
+	"iter"
+	"net/netip"
+	"slices"
+	"testing"
+)
+
+func TestPrefixSetPrefixSeq(t *testing.T) {
+	tests := []struct {
+		add  []netip.Prefix
+		want []netip.Prefix
+	}{
+		{pfxs(), pfxs()},
+		{pfxs("::0/128"), pfxs("::0/128")},
+		{pfxs("::0/128", "::1/128"), pfxs("::0/128", "::1/128")},
+		{pfxs("::0/127", "::0/128"), pfxs("::0/127", "::0/128")},
+		{pfxs("::0/126", "::0/127"), pfxs("::0/126", "::0/127")},
+		{pfxs("::0/1", "::0/128"), pfxs("::0/1", "::0/128")},
+		{pfxs("0::0/127", "::0/128", "::1/128"), pfxs("0::0/127", "::0/128", "::1/128")},
+		{pfxs("0::0/127", "::0/128", "::2/128"), pfxs("0::0/127", "::0/128", "::2/128")},
+		{pfxs("1.2.3.0/24", "1.2.3.4/32"), pfxs("1.2.3.0/24", "1.2.3.4/32")},
+	}
+	for _, tt := range tests {
+		psb := &PrefixSetBuilder{}
+		for _, p := range tt.add {
+			psb.Add(p)
+		}
+		ps := psb.PrefixSet()
+		seq := ps.PrefixSeq()
+		checkPrefixSeq(t, ps.PrefixSeq(), tt.want)
```

Comment:
I think we want `checkPrefixSeq(t, seq, tt.want)`

Reply:
Sorry, missed this when I added `checkYieldFalse`. Fixed in the latest commit.